### PR TITLE
DM-34328: fix collection sorting in query results and scripts

### PR DESCRIPTION
**Pull request by TallJimbo**
## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/daf/butler/script/pruneCollection.py [hunk 1]
```diff
@@ -116,6 +116,8 @@ def addCollection(name: str) -> None:
         for name in collectionNames:
             addCollection(name)
 
+        collections = {k: collections[k] for k in sorted(collections.keys())}
+
```
Comment (n8pease) [inline python/lsst/daf/butler/script/pruneCollection.py:120]: it looks like this line sorts the `collections` dict by its keys, but then never uses it sequentially (and also it's not an ordered dict). Am I misunderstanding it, or is it not needed?

Comment (TallJimbo) [reply python/lsst/daf/butler/script/pruneCollection.py:120]: Usage is in the lists defined in lines 137-139.  `dict` itself now preserves insertion order in Python; `OrderedDict` is I think still around only for some edge case different behavior and backwards compatibility.

Comment (n8pease) [reply python/lsst/daf/butler/script/pruneCollection.py:120]: oh, of course. I didn't expand the view down. Thx.

**Comment by n8pease** [comment]
LGTM

**Pull request merged**